### PR TITLE
feat(typography): add tokens and update shim

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: CDE-778

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/angular/src/layout/nav/_links.clarity.scss [hunk 1]
```diff
@@ -4,11 +4,37 @@
...
+      variables.$clr-p1-line-height,
+      variables.$clr-use-custom-properties
+    );
+    @include mixins.css-var(
+      letter-spacing,
+      clr-link-letter-spacing,
+      variables.$clr-p1-letter-spacing,
+      variables.$clr-use-custom-properties
+    );
+  }
```
Comment (valentin-mladenov) [inline projects/angular/src/layout/nav/_links.clarity.scss:37]: Just make sure we need those link changes. Typography and Buttons design specifications are contradicting.

**Review by Jinnie — APPROVED** [review]
The sooner we have this merged, the sooner dependent tasks will start facing and resolving potential conflicts and raising alerts for problems if any.

File: projects/angular/src/layout/nav/_links.clarity.scss (hunk 1 above)
Comment (dtsanevmw) [reply projects/angular/src/layout/nav/_links.clarity.scss]: Discussed offline and decided to use `inherit` as default which will result in the same values but keep the exposed properties.

**Review by valentin-mladenov — APPROVED** [review]
Perfect

**Pull request merged**